File: OSandDS/data structures/C++/Common_Quest1.cpp

#include <iostream>
#include <unordered_map>
#include <algorithm>
#include <cctype>

bool is_isogram(std::string text) {
    std::transform(text.begin(), text.end(), text.begin(), [](unsigned char c){ return std::tolower(c); });
    std::unordered_map<char, int> unique_dict;
    for (char i_char : text) {
        if (std::isalpha(i_char)) {
            if (unique_dict.find(i_char) == unique_dict.end()) {
                unique_dict[i_char] = 1;
            } else {
                return false;
            }
        }
    }
    return true;
}

int main() {
    std::string text;
    std::cout << "Введите текст: ";
    std::cin >> text;

    if (is_isogram(text)) {
        std::cout << "Введенный текст является изограммой" << std::endl;
    } else {
        std::cout << "Введенный текст не является изограммой" << std::endl;
    }

    return 0;
}

File: OSandDS/data structures/Python/Common_Quest1.py
# Обязательное задание. Напишите программу, которая проверяет, является ли введённая строка
# изограммой. Строка состоит из множества латинских букв в разном регистре, пробелов и знаков
# препинания. Причём в изограмме пробелы или знаки препинания могут повторяться.

def is_isogram(text: str) -> bool:
    lower_text = text.lower()
    unique_dict = dict()
    for i_char in lower_text:
        if i_char.isalpha():
            if i_char not in unique_dict:
                unique_dict[i_char] = 1
            else:
                return False
    return True


if is_isogram(input("Введите текст: ")):
    print("Введенный текст является изограммой")
else:
    print("Введенный текст не явялется изограммой")

